### ISSUE TITLE
Enable dataflow-beam-import integration tests

### ISSUE THREAD

**Issue by dmitry-fa** [googleapis/java-bigtable-hbase#2623]
There are two integration tests which never executed:

- `bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/sequencefiles/EndToEndIT.java`
- `bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/it/CloudBigtableBeamITTest.java`

They could be added to the regular execution to increase the coverage.

**Comment by dmitry-fa** [comment]
@igorbernstein2, do you agree that these tests should be added to execution?

**Comment by igorbernstein2** [comment]
Yep, but I think you mind need to check with @kolea2

**Comment by kolea2** [comment]
Thanks for looking into this @dmitry-fa!

What is required from an infrastructure standpoint to get these tests running/what did you have to do to get them running locally?

For example, I see that CloudBigtableBeamITTest requires some extra params:
https://github.com/googleapis/java-bigtable-hbase/blob/master/bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/it/CloudBigtableBeamITTest.java#L84-L87

Are these tests activated via a maven profile?

**Comment by dmitry-fa** [comment]
@kolea, I managed to run them locally. Yes, these tests require extra parameters, I'll take care to add them to kokoro scripts to make them running in CICD and document in readme. I'm thinking about how to do this in the best way.

**Comment by kolea2** [comment]
Awesome. Some thoughts I have:

1. We could remove the bucket as a parameter for the test, and dynamically create and clean up a test bucket per test
2. Is this test already covered in the active IT profiles getting tested? Or do we need to add it to the kokoro config? For clarity, it may be easier to separate them out into a new kokoro job then. I can do that - LMK what you think

**Comment by dmitry-fa** [comment]
> We could remove the bucket as a parameter for the test, and dynamically create and clean up a test bucket per test

agree. I think `zoneId` argument can be made optional as well.

> Is this test already covered in the active IT profiles getting tested? Or do we need to add it to the kokoro config? For clarity, it may be easier to separate them out into a new kokoro job then. I can do that - LMK what you think

I'm goint to include it into the standard  `-Penable-integration-tests` profile. I will let you know if I have any difficulties.

**Comment by kolea2** [comment]
> agree. I think `zoneId` argument can be made optional as well.

SGTM

@dmitry-fa please see the current IT setup in the kokoro configs (which are different than the other client libraries):
https://github.com/googleapis/java-bigtable-hbase/blob/master/.kokoro/build.sh#L54
profiles passed in: https://github.com/googleapis/java-bigtable-hbase/blob/master/.kokoro/presubmit/integration.cfg#L11

I think it makes sense to create a new maven profile for beam tests, and add it to this kokoro config.

**Comment by dmitry-fa** [comment]
Yes, agree. Actually there are three profiles for dataflow:
- `bigtableDataflowIntegrationTest` for CloudBigtableIOIntegrationTest.java (hbase-beam)
- `beamIntegrationTest` for CloudBigtableBeamITTest.java (beam-import)
- `sequencefileIntegrationTest` for sequencefiles/EndToEndIT.java

I will add them all.

**Comment by kolea2** [comment]
Sounds good. At a quick glance, I would also change this to be more specific: https://github.com/googleapis/java-bigtable-hbase/blob/master/bigtable-dataflow-parent/bigtable-beam-import/pom.xml#L322

**Comment by dmitry-fa** [comment]
@kolea2 
> We could remove the bucket as a parameter for the test, and dynamically create and clean up a test bucket per test

To do that we need to add a dependency on the Storage, this is not a trivial task because bigtable project includes some ancient versions of the Storage. This optimization could be done in a separate PR and include cleanup of dependencies.

**Comment by igorbernstein2** [comment]
Ideally resource creation should be minimized during a test. It adds noise and flakiness. Whats rationale for removing these params?

**Comment by dmitry-fa** [comment]
> Whats rationale for removing these params?

For a user, it's easier to configure the tests that require gs://my-bucket parameters to run them locally.

**Comment by kolea2** [comment]
> > Whats rationale for removing these params?
> 
> For a user, it's easier to configure the tests that require gs://my-bucket parameters to run them locally.

My thought was that we'd create a test bucket per test run (and subsequent cleanup) for that reason as well as to account for tests running in parallel. @igorbernstein2 does that make sense to you?

**Comment by igorbernstein2** [comment]
Bucket creation is fairly expensive operation. It would make more sense to vary the path per test run, but keep the bucket the same

**Comment by kolea2** [comment]
SGTM

**Comment by dmitry-fa** [comment]
Now the beam-import tests expect `-Dgoogle.dataflow.stagingLocation=gs://[google storage bucket]` and `-Dcloud.test.data.folder=gs://...` parameters. I will use unified `-Dgoogle.dataflow.gcsPath=gs://[google storage path]` and  create temporary subfolders per test.

**Comment by dmitry-fa** [comment]
@kolea2, could you create a bucket, please.
